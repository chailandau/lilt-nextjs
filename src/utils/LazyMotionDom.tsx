import { LazyMotion, domAnimation } from 'framer-motion';
import { FC, ReactNode } from 'react';

interface LazyMotionDomProps {
    children: ReactNode;
}

const LazyMotionDom: FC<LazyMotionDomProps> = ({ children }) => (
    <LazyMotion features={domAnimation}>
        {children}
    </LazyMotion>
);

export default LazyMotionDom;
