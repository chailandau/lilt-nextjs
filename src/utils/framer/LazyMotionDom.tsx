import { LazyMotion, domAnimation } from 'framer-motion';
import { FC, ReactNode } from 'react';

interface LazyMotionDomProps {
    children: ReactNode;
}

const LazyMotionDom: FC<LazyMotionDomProps> = ({ children }) => (
    <LazyMotion features={domAnimation} strict>
        {children}
    </LazyMotion>
);

export default LazyMotionDom;
