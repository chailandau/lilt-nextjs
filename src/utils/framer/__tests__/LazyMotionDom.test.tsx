import {
    testAxeViolations,
    testMatchesSnapshot,
    testRenderChildren
} from '../../testHelpers';
import LazyMotionDom from '../LazyMotionDom';

const LazyMotionHOC = () => (
    <LazyMotionDom>
        <div>Test</div>
    </LazyMotionDom>
);

const lazyMotionComponent = {
    component: <LazyMotionHOC />
};

describe('LazyMotionDom', () => {
    testRenderChildren(lazyMotionComponent);
    testAxeViolations(lazyMotionComponent);
    testMatchesSnapshot(lazyMotionComponent);
});
