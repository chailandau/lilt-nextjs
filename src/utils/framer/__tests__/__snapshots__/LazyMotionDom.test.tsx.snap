// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyMotionDom matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Test
  </div>
</DocumentFragment>
`;
